src/Host/Broker/Impl/RemoteUri/RemoteUriHelper.cs

```diff
@@ -5,37 +5,88 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Net;
+using System.Net.WebSockets;
+using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Http;
 using Microsoft.Common.Core;
 using Microsoft.R.Host.Protocol;
 
 namespace Microsoft.R.Host.Broker.RemoteUri {
     public class RemoteUriHelper {
+
+        private static async Task DoWebSocketReceiveSendAsync(WebSocket receiver, WebSocket sender, CancellationToken ct) {
+            if (receiver == null || receiver == null) {
+                return;
+            }
+
+            ArraySegment<byte> receiveBuffer = WebSocket.CreateServerBuffer(65335);
```

Comment:
I'm not entirely sure what this method does, but I am pretty sure that it's not really necessary - you can just `new` up an array and wrap it in an array segment. I'm worried that going through the method might result in a buffer shared with something else, and potential races.